### PR TITLE
Add PR karma

### LINKED ISSUE #72: Add pr karma script
See https://gist.github.com/noisecapella/c7178e728c36a84e2dee0d2dffba1401

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #72 

#### What's this PR do?
 - Adds a command to calculate pull request karma. Example usage: `karma 2017-11-09` to calculate karma between that date and today.
 - Adds `bot_local.py` which runs a single command on a particular channel, but without actually connecting to slack.

#### How should this be manually tested?
 - Get a github access token. You can create one in the settings for your account. Use it for `GITHUB_ACCESS_TOKEN`
 - Run `./bot_local.py karma 2017-11-09`. You should see a list of people and their karma score, sorted from biggest to smallest.

### CODE REVIEW

File: github.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    data = run_query(github_access_token, KARMA_QUERY)
+
+    karma = defaultdict(lambda: 0)
+    for repository in data['data']['organization']['repositories']['nodes']:
+        # Keep track if any dates fall outside the range. If none do and we're at the max limit for number of PRs,
+        # we need to paginate (but instead we'll just raise an exception for now).
+        outside_range = None
+        for pull_request in repository['pullRequests']['nodes']:
+            merged_at = parse(pull_request['updatedAt']).date()
+            updated_at = parse(pull_request['mergedAt']).date()
```
Comment (gsidebo) [inline github.py]: Looks like you've flipped these variable names

File: bot.py [hunk 2]
```diff
@@ -79,28 +82,43 @@ def in_script_dir(file_path):
...
+    """Get required environment variables"""
+    required = ('SLACK_ACCESS_TOKEN', 'BOT_ACCESS_TOKEN', 'GITHUB_ACCESS_TOKEN')
+
+    for key in required:
+        value = os.environ.get(key)
+        if not value:
+            raise Exception("Missing {}".format(key))
+    return {
+        key: os.environ.get(key) for key in required
+    }
```
[the remark this answers is not in the document]
Comment (noisecapella) [reply bot.py]: Sure, I also changed the second line to be a dict comprehension:

    env_dict = {key: os.environ.get(key, None) for key in required_keys}

File: github.py (hunk 1 above)
Comment (noisecapella) [reply github.py]: Ah good catch

**Comment by noisecapella** [comment]
Comments addressed

File: bot_local.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    if len(sys.argv) < 3:
+        raise Exception("Expected arguments: project_name command arg1 arg2...")
+
+    _, project_name, *words = sys.argv
+
+    repos_info = load_repos_info()
+    try:
+        repo_info = [repo_info for repo_info in repos_info if repo_info.name == project_name][0]
+    except IndexError:
+        raise Exception("Unable to find channel with name {}".format(project_name))
```
Comment (gsidebo) [inline bot_local.py]: I think this message should be something like `Unable to find repo with name {}` or `Repo "{}" either doesn't exist or has not been configured to use the release script (see repos_info.json)`

**Comment by noisecapella** [comment]
I'll change `bot_local` to take a channel name instead of a repo name. The intention is to act as if the bot received a command in a particular channel. The channel shouldn't matter for the karma command, I would imagine we would usually use it in odl-engineering but it doesn't do any checks on what channel it's in

**Pull request merged**